script.js

```diff
@@ -220,21 +223,33 @@ window.addEventListener('load', () => {
       e.preventDefault();
       return;
     }
-    const name = document.getElementById('name');
+
     if (!name.validity.valid) {
       // submit form
       document.getElementById('error').innerHTML = 'Invalid Name';
       e.preventDefault();
       return;
     }
-    const text = document.getElementById('text');
+
     if (!text.validity.valid) {
       // submit form
       document.getElementById('error').innerHTML = 'Invalid Text';
       e.preventDefault();
       return;
     }
-    // show errors
+
+    const form = document.getElementById('contact-form');
+    const myFormData = new FormData(form);
+    const formDataObj = Object.fromEntries(myFormData.entries());
+    localStorage.setItem('contactform', JSON.stringify(formDataObj));
+    // submit form
     document.getElementById('contact-form').submit();
   });
+
+  const contactFormData = JSON.parse(
+    localStorage.getItem('contactform') ?? '{}',
+  );
+  name.value = contactFormData.name ?? '';
+  email.value = contactFormData.email ?? '';
+  text.value = contactFormData.message ?? '';
 });
```

Comment:
- [ ] Hi, kindly add an on load function which pulls the data from local storage and populates it into the input fields upon reloading the page. In the current scenario the data is not displayed

Reply:
Thank you @Munsa1. 

Do you mean we should put the code above in an onload event? if yes, that is what we have done, and the data actually populates when you reload the page after filling the form.

could you clarify more on what we need to do?